java/server/src/org/openqa/selenium/grid/sessionmap/jdbc/JdbcBackedSessionMap.java

```diff
@@ -0,0 +1,175 @@
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.grid.sessionmap.jdbc;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.ImmutableCapabilities;
+import org.openqa.selenium.NoSuchSessionException;
+import org.openqa.selenium.grid.config.Config;
+import org.openqa.selenium.grid.config.ConfigException;
+import org.openqa.selenium.grid.data.Session;
+import org.openqa.selenium.grid.log.LoggingOptions;
+import org.openqa.selenium.grid.sessionmap.SessionMap;
+import org.openqa.selenium.internal.Require;
+import org.openqa.selenium.json.Json;
+import org.openqa.selenium.remote.SessionId;
+import org.openqa.selenium.remote.tracing.Tracer;
+
+import java.io.Closeable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.logging.Logger;
+
+
+public class JdbcBackedSessionMap extends SessionMap implements Closeable {
+
+  private static final Json JSON = new Json();
+  private static final Logger LOG = Logger.getLogger(JdbcBackedSessionMap.class.getName());
+  private final Connection connection;
+  private final String  tableName;
+  private final String sessionIdCol;
+  private final String sessionCapsCol;
+  private final String sessionUriCol;
+
+
+  public JdbcBackedSessionMap(Tracer tracer, Connection jdbcConnection, String sessionTableName, String sessionIdCol, String sessionCapsCol, String sessionUriCol)  {
+    super(tracer);
+
+    Require.nonNull("JDBC Connection Object", jdbcConnection);
+    Require.nonNull("Table name", sessionTableName);
+    Require.nonNull("Session id column name", sessionIdCol);
+    Require.nonNull("Session caps column name", sessionCapsCol);
+    Require.nonNull("Session URI column name", sessionUriCol);
+
+    this.connection = jdbcConnection;
+    this.tableName = sessionTableName;
+    this.sessionIdCol = sessionIdCol;
+    this.sessionCapsCol = sessionCapsCol;
+    this.sessionUriCol = sessionUriCol;
+  }
+
+  public static SessionMap create(Config config) {
+    Tracer tracer = new LoggingOptions(config).getTracer();
+    JdbcSessionMapOptions sessionMapOptions = new JdbcSessionMapOptions(config);
+    String tableName = sessionMapOptions.getJdbcTableName();
+    String sessionIdColName = sessionMapOptions.getJdbcSessionIdColName();
```

Comment:
All of these should be hard-coded now, so there's no need to look them up in the options :)